### PR TITLE
feat(load): WHOOP API v2 loader with OAuth2

### PR DESCRIPTION
**Pull request by ErikBjare**
## Why

Whoop physiology in the QS export has been frozen at 2026-05-13 (ErikBjare/alice#65). Root cause: `config.toml` points at a static CSV export dir downloaded on 05-13 — there was never any OAuth to expire. This adds a live WHOOP API v2 integration so the data keeps advancing without manual re-downloads.

## What

- **`load/whoop_api.py`** — new module:
  - OAuth2 authorization-code flow (`python -m quantifiedme.load.whoop_api auth`): one-shot localhost redirect server, browser consent, token persisted to platformdirs with 0600 perms.
  - **Rotation-safe refresh**: WHOOP invalidates the old refresh token on every refresh, so the rotated token is written to disk immediately.
  - Paginated fetchers (`nextToken`, limit 25, 429 backoff) for cycles / recoveries / sleeps / workouts.
  - Incremental on-disk JSON cache keyed by record id, re-fetching a 14-day overlap window to pick up late score revisions — daily export runs stay fast.
  - Mappers producing **the same DataFrame schemas as the CSV loaders** (verified column-for-column against `_load_sleep_standard` / `_load_cycles_standard` / `_load_workouts_standard`), incl. wake-date indexing convention and kJ→kcal conversion.
  - CLI: `auth` / `status` / `fetch`.
- **`load/whoop.py`** — public loaders dispatch to the API when a token exists on the machine; file exports remain the fallback. API errors are deliberately *not* swallowed into a file fallback — silent staleness is exactly the failure mode this fixes. Journal loader stays file-based (API doesn't expose journal entries).
- Credentials from `WHOOP_CLIENT_ID`/`WHOOP_CLIENT_SECRET` env vars or `.env.whoop` at repo root (now gitignored via `.env.*`).
- `requests` promoted to a direct dependency.

## Tests

25 new tests (fixtures match the official v2 response schemas): mapping incl. tz-offset wake-date edge cases, nap/unscored exclusion, refresh-token rotation persistence, pagination param casing (`next_token` in response vs `nextToken` in query), incremental cache upsert, and dispatch. Existing whoop CSV tests pinned to file dispatch so a real token on a dev machine doesn't flip them. All 50 whoop tests pass.

### CODE REVIEW

**Comment by ErikBjare** [comment]
@greptileai review

File: src/quantifiedme/load/whoop_api.py [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+                "timestamp": _local_date(r["end"], r.get("timezone_offset", "Z")),
+                "score": score.get("sleep_performance_percentage"),
+                "duration": ms(asleep, unit="ms"),
+                "time_in_bed": ms(stages["total_in_bed_time_milli"], unit="ms"),
+                "efficiency": score.get("sleep_efficiency_percentage"),
+                "consistency": score.get("sleep_consistency_percentage"),
+                "debt": (score.get("sleep_needed") or {}).get(
+                    "need_from_sleep_debt_milli", 0
+                )
+                / 60_000,
```
[the remark this answers is not in the document]
Comment (ErikBjare) [reply src/quantifiedme/load/whoop_api.py]: Fixed in fd8c930

File: src/quantifiedme/load/whoop_api.py [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+            {
+                "timestamp": date,
+                "recovery": score.get("recovery_score"),
+                "resting_hr": score.get("resting_heart_rate"),
+                "hrv": score.get("hrv_rmssd_milli"),
+                "skin_temp": score.get("skin_temp_celsius"),
+                "spo2": score.get("spo2_percentage"),
+                "strain": cycle_score.get("strain"),
+                "energy_kcal": kilojoule * KCAL_PER_KILOJOULE if kilojoule else None,
+            }
```
[the remark this answers is not in the document]
Comment (ErikBjare) [reply src/quantifiedme/load/whoop_api.py:426]: Fixed in fd8c930

File: src/quantifiedme/load/whoop_api.py [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+        kilojoule = score.get("kilojoule")
+        rows.append(
+            {
+                "start": start,
+                "end": end,
+                "duration": end - start,
+                "activity": r.get("sport_name"),
+                "strain": score.get("strain"),
+                "energy_kcal": kilojoule * KCAL_PER_KILOJOULE if kilojoule else None,
+                "max_hr": score.get("max_heart_rate"),
```
[the remark this answers is not in the document]
Comment (ErikBjare) [reply src/quantifiedme/load/whoop_api.py:446]: Fixed in fd8c930

File: src/quantifiedme/load/whoop_api.py [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+    ]
+    if not rows:
+        return pd.DataFrame(columns=columns)
+    return pd.DataFrame(rows)[columns].sort_values("start").reset_index(drop=True)
+
+
+def _to_daily_df(rows: list[dict[str, Any]]) -> pd.DataFrame:
+    df = pd.DataFrame(rows)
+    if df.empty:
+        return df
```
[the remark this answers is not in the document]
Comment (ErikBjare) [reply src/quantifiedme/load/whoop_api.py]: Fixed in fd8c930
